spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskServiceTests.java

```diff
@@ -72,364 +72,402 @@
-		assertEquals("globalvalue", request.getDefinition().getProperties().get("globalkey"));
-		assertNull(request.getDefinition().getProperties().get("globalstreamkey"));
-	}
+		@Test
+		@DirtiesContext
+		public void createSimpleTask() {
+			initializeSuccessfulRegistry(appRegistry);
+			taskService.saveTaskDefinition("simpleTask", "AAA --foo=bar");
+			verifyTaskExistsInRepo("simpleTask", "AAA --foo=bar", taskDefinitionRepository);
+		}
 
-	@Test
-	@DirtiesContext
-	public void executeComposedTaskWithLabels() {
-		String dsl = "t1: AAA && t2: BBB";
-		initializeSuccessfulRegistry();
-		taskService.saveTaskDefinition("seqTask", dsl);
-		when(taskLauncher.launch(anyObject())).thenReturn("0");
-		Map<String, String> properties = new HashMap<>();
-		properties.put("app.seqTask.t1.timestamp.format", "YYYY");
-		properties.put("app.composed-task-runner.interval-time-between-checks", "1000");
-		assertEquals(1L, this.taskService.executeTask("seqTask", properties, new LinkedList<>()));
-		ArgumentCaptor<AppDeploymentRequest> argumentCaptor = ArgumentCaptor.forClass(AppDeploymentRequest.class);
-		verify(this.taskLauncher, atLeast(1)).launch(argumentCaptor.capture());
-
-		AppDeploymentRequest request = argumentCaptor.getValue();
-		assertEquals("seqTask", request.getDefinition().getProperties().get("spring.cloud.task.name"));
-		assertTrue(request.getDefinition().getProperties().containsKey("composed-task-properties"));
-		assertEquals("app.seqTask-t1.app.AAA.timestamp.format=YYYY", request.getDefinition().getProperties().get("composed-task-properties"));
-		assertTrue(request.getDefinition().getProperties().containsKey("interval-time-between-checks"));
-		assertEquals("1000", request.getDefinition().getProperties().get("interval-time-between-checks"));
-	}
+		@Test
+		@DirtiesContext
+		public void executeSingleTaskTest() {
+			initializeSuccessfulRegistry(appRegistry);
+			when(taskLauncher.launch(anyObject())).thenReturn("0");
+			assertEquals(1L, this.taskService.executeTask(TASK_NAME_ORIG, new HashMap<>(), new LinkedList<>()));
+		}
 
-	@Test
-	@DirtiesContext
-	public void executeTaskWithNullIDReturnedTest() {
-		initializeSuccessfulRegistry();
-		boolean errorCaught = false;
-		when(this.taskLauncher.launch(anyObject())).thenReturn(null);
-		try {
-			taskService.executeTask(TASK_NAME_ORIG, new HashMap<>(), new LinkedList<>());
+		@Test
+		@DirtiesContext
+		public void executeMultipleTasksTest() {
+			initializeSuccessfulRegistry(appRegistry);
+			when(taskLauncher.launch(anyObject())).thenReturn("0");
+			assertEquals(1L, this.taskService.executeTask(TASK_NAME_ORIG, new HashMap<>(), new LinkedList<>()));
+			assertEquals(2L, this.taskService.executeTask(TASK_NAME_ORIG, new HashMap<>(), new LinkedList<>()));
 		}
-		catch (IllegalStateException ise) {
-			errorCaught = true;
-			assertEquals("Deployment ID is null for the task:myTask_ORIG", ise.getMessage());
+
+		@Test(expected = IllegalStateException.class)
```

Comment:
A nice way to catch exception and check exception message afterword  is to use ExpectedException.   As shown here:https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSkipperStreamServiceTests.java#L134-142